### PR TITLE
Handle nusmods `ApiException` and invalid academic year

### PR DESCRIPTION
**Pull request by shenyih0ng**

### CODE REVIEW

**Review by huzaifa1712 — CHANGES_REQUESTED** [review]
An input like `1966-1967` or `2023-2024` follows the format correctly but the NUSMods API doesn't have mods for that AY.
Same for an input like `2020-2022` which has the right format but isn't a valid AY.

Should we add something to the error dialog such as `The academic year specified should be consecutive years e.g 2022-2023` and also `Do note that the academic year may have the correct format but might be too far into the past or future.` ?

Sounds a bit obvious but just to protect against them flagging as a bug..

**Comment by shenyih0ng** [comment]
> An input like `1966-1967` or `2023-2024` follows the format correctly but the NUSMods API doesn't have mods for that AY. Same for an input like `2020-2022` which has the right format but isn't a valid AY.
> 
> Should we add something to the error dialog such as `The academic year specified should be consecutive years e.g 2022-2023` and also `Do note that the academic year may have the correct format but might be too far into the past or future.` ?

Yes i'm currently doing that by adding it in the UG since it will be quite a long description. Will link it on the alert window

**Review by huzaifa1712 — APPROVED** [review]

File: docs/UserGuide.md [hunk 1]
```diff
@@ -232,28 +234,41 @@ Traverse/Scroll **down**: `<Down>`
...
+    Since, `CLIMods` relies on the [`nusmods` API](https://api.nusmods.com/v2) for module data, do note that legacy module data may not be available. For example, since `nusmods` no longer provide module data before `2018`, specifying `2017-2018` or any previous academic years will be treated as an invalid academic year.
 
-Ans: CLIMods requires internet access because we fetch module data from NUSMods API.  Please ensure that you have
-internet access before launching CLIMods.
+    <div markdown="span" class="alert alert-primary">
+    :bulb: **Tip**
+    
+    - Valid academic year examples: `2022-2023`, `2018-2019`
+    - Invalid academic year examples: `2022-2022`, `2023-2022`, `22-23`
+    </div>
```
Comment (shenyih0ng) [inline docs/UserGuide.md:274]: @huzaifa1712 let me know if this is okay

Comment (huzaifa1712) [reply docs/UserGuide.md:274]: LGTM! Thanks for doing this

**Review by SeekSaveServe — APPROVED** [review]
LGTM! works on my end

**Review by sarrrdin — APPROVED** [review]
Works well on my device! LGTM 👍 Thank you @shenyih0ng!

**Pull request merged**